app/javascript/src/components/Dashboard/Contacts/Table/index.jsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+
+import { MenuHorizontal } from "neetoicons";
+import { Table as NeetoUITable, Typography, Avatar } from "neetoui";
+
+import { CONTACT_TABLE_COLUMN_DATA } from "../constants";
+import { stringToDate } from "../utils";
+
+const Table = ({
+  selectedContactIds,
+  setSelectedContactIds,
+  contacts = [],
+}) => {
+  const modifyContactDateFormat = contacts =>
+    contacts.map(contact => ({
+      ...contact,
+      created_at: stringToDate(contact["created_at"]),
+    }));
+  const renderMenuColumn = () => <MenuHorizontal size={16} />;
+  const renderProfileColumn = (name, { role, profile_pic }) => (
+    <div className="flex flex-row items-center">
+      <Avatar
+        className="mr-2"
+        size="large"
+        user={{ name, imageUrl: profile_pic }}
+      />
+      <div className="flex flex-col">
+        <Typography style="h5">{name}</Typography>
+        <Typography style="body3">{role}</Typography>
+      </div>
+    </div>
+  );
+  const buildContactColumns = columnsWithNoRenderedComponents =>
+    columnsWithNoRenderedComponents.map(column => {
+      if (column["dataIndex"] === "name") {
+        return { ...column, render: renderProfileColumn };
+      } else if (column["dataIndex"] === "icon_button") {
+        return { ...column, render: renderMenuColumn };
+      }
+
+      return column;
+    });
+  return (
+    <div className="contact-table-height table-striped-rows w-full">
+      <NeetoUITable
+        allowRowClick
+        rowSelection
+        columnData={buildContactColumns(CONTACT_TABLE_COLUMN_DATA)}
+        currentPageNumber={1}
+        defaultPageSize={9}
+        rowData={modifyContactDateFormat(contacts)}
+        selectedRowKeys={selectedContactIds}
+        onRowSelect={selectedRowKeys => setSelectedContactIds(selectedRowKeys)}
+      />
+    </div>
+  );
+};
```

Comment:
The component is supposed to be the presentation layer. So let's try our level best to adhere to that. So whatever utils can moved out into utils.js or respective files.

Reply:
Ok, Yedhin I will move the function `modifyContactDateFormat` to utils.js. Thanks for pointing this out.
Although I am skeptic about moving the other functions into` utils.js` since I would have to use JSX components inside a js file. If I were to rename `utils.js` to `utils.jsx`, wouldn't it violate the naming standards for JSX components?

Reply:
>If I were to rename utils.js to utils.jsx, wouldn't it violate the naming standards for JSX components?

No. We can use the `.jsx` extensions whenever we have to handle some JSX within that file. Refer: https://github.com/bigbinary/bigbinary-website/issues/2453